packages/amplify-e2e-core/src/init/adminUI.ts

```diff
@@ -0,0 +1,41 @@
+import { setupAmplifyAdminUI, getAmplifyBackendJobStatus } from '../utils/sdk-calls';
+
+export async function enableAdminUI(appId: string, envName: string, region: string) {
+  const setupAdminUIJobDetails = await setupAmplifyAdminUI(appId, region);
+
+  const jobCompletionDetails = await pollUntilDone(setupAdminUIJobDetails.JobId, appId, envName, region, 2 * 1000, 2000 * 1000);
+
+  if (jobCompletionDetails.Status === 'FAILED') {
+    throw new Error('Setting up Admin UI failed');
+  }
+}
+
+// interval is how often to poll
+// timeout is how long to poll waiting for a result (0 means try forever)
+
+async function pollUntilDone(jobId: string, appId: string, envName: string, region: string, interval: number, timeout: number) {
```

Comment:
we really shouldn't duplicate this function...

Reply:
I don't think we should be sharing context, helpers for the actual CLI code and the E2E tests. The way the SDK calls are being made are different. When we move the helpers to it's own package as we've discussed offline, we could maybe have a sample poller.